hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/TokenTupleUtils.java

```diff
@@ -63,4 +68,107 @@ public static Tuple keyTupleFor(@NonNull final Key key) {
                 key.ecdsaSecp256k1OrElse(Bytes.EMPTY).toByteArray(),
                 headlongAddressOf(key.delegatableContractIdOrElse(ZERO_CONTRACT_ID)));
     }
+
+    /**
+     * Returns a tuple containing the response code, fixedFees, fractionalFees and the royaltyFees for the token
+     * @param token the token to get the fees for
+     * @return Tuple encoding of the arrays of fixedFees, fractionalFees and royaltyFees
+     */
+    @NonNull
+    public static Tuple feesTupleFor(final int responseCode, @NonNull final Token token) {
+        final var fixedFees = fixedFeesTupleListFor(token);
+        final var fractionalFees = fractionalFeesTupleListFor(token);
+        final var royaltyFees = royaltyFeesTupleListFor(token);
+        return Tuple.of(
+                responseCode,
+                fixedFeesTupleListFor(token).toArray(new Tuple[fixedFees.size()]),
+                fractionalFeesTupleListFor(token).toArray(new Tuple[fractionalFees.size()]),
+                royaltyFeesTupleListFor(token).toArray(new Tuple[royaltyFees.size()]));
+    }
+
+    /**
+     * Returns a list of Tuples defined by the {@code FixedFee} structs
+     * {@see https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L236 }
+     * @param token the token to get the fixed fees for
+     * @return Tuple encoding of the FixedFee
+     */
+    @SuppressWarnings("DataFlowIssue")
+    @NonNull
+    private static List<Tuple> fixedFeesTupleListFor(@NonNull final Token token) {
+        //
+        assert token.customFees() != null;
+        return token.customFees().stream()
+                .filter(CustomFee::hasFixedFee)
+                .filter(fee -> fee.fixedFee() != null)
+                .map(fee -> Tuple.of(
+                        fee.fixedFee().amount(),
```

Comment:
Would you be able to drop the second `.filter()` by using `fee.fixedFeeOrThrow()` here?

Reply:
I'll make this fix in the next one in the series.